system/Validation/Rules.php

```diff
@@ -304,9 +304,9 @@ public function required_with($str = null, string $fields, array $data): bool
 		// If the field is present we can safely assume that
 		// the field is here, no matter whether the corresponding
 		// search field is present or not.
-		$present = $this->required($data[$str] ?? null);
+		$present = $this->required($str ?? '');
```

Comment:
Without going back and digging into the logic, are you sure this is correct? Using `$str` instead of `$data[$str]` would check that the current field is required, not that the `with` field is required...

Reply:
Yes. `$str` is field value. `$data` is Post data. so `$data[$str]` will never get the value.
show demo:
`protected $validationRules    = [
        'password'     => 'required|min_length[8]',
        'password_confirm' => 'matches[password]'
    ];`

`curl -X POST \
  localhost/users \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -H 'cache-control: no-cache' \
  -d 'password=toto&password_confirm=toto'`

`$str` is toto.  `$data['toto']` is null. so `$present` has always been false.

Reply:
I tested is no problem.